cmd/kogito/command/deploy/deploy_service.go

```diff
@@ -278,3 +319,149 @@ func (i *deployCommand) Exec(cmd *cobra.Command, args []string) (err error) {
 
 	return nil
 }
+
+// matchKogitoResources will verify if the given parameter is a expect drl or dmn file.
+// Accepted values are, supported file extensions as it is, directory or url
+// See supportedExtensions env
+// return the file content as io.Reader if the provided file or url is valid.
+func matchKogitoResources(resource string) (io.Reader, *buildv1.BinaryBuildRequestOptions, error) {
+	log := context.GetDefaultLogger()
+	options := &buildv1.BinaryBuildRequestOptions{}
+	var fileName string
+	// from url, do only basic check, for example, if the url has the suffix that's match the allowed ones
+	// if found download it and save on /tmp
+	switch {
+	case strings.HasPrefix(resource, "http"):
+		parsedURL, err := url.ParseRequestURI(resource)
+		if err != nil {
+			return nil, nil, fmt.Errorf("kogito resource is not a valid URL, received %s", resource)
+		}
+
+		hasExtension := false
+		ff := strings.Split(parsedURL.Path, "/")
+		fileName = strings.Join(strings.Fields(ff[len(ff)-1]), "")
+
+		if isSuffixSupported(fileName) {
+			hasExtension = true
+		}
+
+		if hasExtension {
```

Comment:
The hasExtension can be removed and content of this condition can be moved under `if isSuffixSupported(fileName) {`

Reply:
+1